File: Lista0/Lista0/zbiorPunktow3D.cpp
#include <iostream>
#include <algorithm>
#include <iterator>
#include <unordered_set>

#include "zbiorPunktow3D.h"
#include "punkt3D.h"

using namespace std;

zbiorPunktow3D::zbiorPunktow3D() {
	unordered_set<punkt3D> s = {};
}

void zbiorPunktow3D::dodajPunkt(punkt3D p) {
	s.insert(p);
}


void zbiorPunktow3D::usunPunkt(const punkt3D& p) {
    if (s.erase(p) == 0) {
        cout << "Nie znaleziono punktu!";
    }
    else {
        s.erase(p);
    }
}

bool zbiorPunktow3D::czyPunktJestWZbiorze(const punkt3D& p) {
    return s.find(p) != s.end();
}

int zbiorPunktow3D::mocZbioru() const {
    return s.size();
}

void zbiorPunktow3D::wyswietl() const {
    cout << "[ ";
    for (const auto& p : s) {
        cout << p;
    }
    cout << "]" << endl;
}

File: lista0java/src/main/java/com/mycompany/lista0java/ZbiorPunktow3D.java
package com.mycompany.lista0java;

import java.util.HashSet;
import java.util.Set;

public class ZbiorPunktow3D {

    private Set<Punkt3D> zbior;

    public ZbiorPunktow3D() {
        zbior = new HashSet<>();
    }

    public void dodajPunkt(Punkt3D p) {
        zbior.add(p);
    }

    public void usunPunkt(Punkt3D p) {
        if (!zbior.remove(p)) {
            System.out.println("Nie znaleziono punktu!");
        }
    }

    public boolean czyPunktJestWZbiorze(Punkt3D p) {
        return zbior.contains(p);
    }

    public int mocZbioru() {
        return zbior.size();
    }

    public void wyswietl() {
        System.out.print("[ ");
        for (Punkt3D p : zbior) {
            System.out.print(p + " ");
        }
        System.out.println("]");
    }
}